[quant][fix] Remove activation_post_process in qat modules

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#42343 [quant][fix] Remove activation_post_process in qat modules**

Summary:
Currently activation_post_process are inserted by default in qat modules, which is not
friendly to automatic quantization tools, this PR removes them.

Test Plan:
regression tests
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22856816](https://our.internmc.facebook.com/intern/diff/D22856816)

jerryzh168 committed:
[quant][fix] Remove activation_post_process in qat modules

Summary:
Currently activation_post_process are inserted by default in qat modules, which is not
friendly to automatic quantization tools, this PR removes them.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fix] Remove activation_post_process in qat modules"


Summary:
Currently activation_post_process are inserted by default in qat modules, which is not
friendly to automatic quantization tools, this PR removes them.

Test Plan:
regression tests
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22856816](https://our.internmc.facebook.com/intern/diff/D22856816)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fix] Remove activation_post_process in qat modules"


Summary:
Currently activation_post_process are inserted by default in qat modules, which is not
friendly to automatic quantization tools, this PR removes them.

Test Plan:
regression tests
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22856816](https://our.internmc.facebook.com/intern/diff/D22856816)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fix] Remove activation_post_process in qat modules"


Summary:
Currently activation_post_process are inserted by default in qat modules, which is not
friendly to automatic quantization tools, this PR removes them.

Test Plan:
regression tests
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22856816](https://our.internmc.facebook.com/intern/diff/D22856816)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fix] Remove activation_post_process in qat modules"


Summary:
Currently activation_post_process are inserted by default in qat modules, which is not
friendly to automatic quantization tools, this PR removes them.

Test Plan:
regression tests
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22856816](https://our.internmc.facebook.com/intern/diff/D22856816)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fix] Remove activation_post_process in qat modules"


Summary:
Currently activation_post_process are inserted by default in qat modules, which is not
friendly to automatic quantization tools, this PR removes them.

Test Plan:
regression tests
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22856816](https://our.internmc.facebook.com/intern/diff/D22856816)

[ghstack-poisoned]

raghuramank100 commented on torch/quantization/quantize.py:
```diff
@@ -330,8 +330,12 @@ def prepare_qat(model, mapping=None, inplace=False):
+
+    propagate_qconfig_(model, qconfig_dict=None)
     _convert(model, mapping, inplace=True)
+    prepare(model, observer_non_leaf_module_list=set(mapping.values()), inplace=True)
```
Can you please fix the doc-strings for convert and _convert as they are not tied to replacing with quantized modules. Perhaps better to say that they swap modules?

jerryzh168 committed:
Update on "[quant][fix] Remove activation_post_process in qat modules"


Summary:
Currently activation_post_process are inserted by default in qat modules, which is not
friendly to automatic quantization tools, this PR removes them.

Test Plan:
regression tests
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22856816](https://our.internmc.facebook.com/intern/diff/D22856816)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fix] Remove activation_post_process in qat modules"


Summary:
Currently activation_post_process are inserted by default in qat modules, which is not
friendly to automatic quantization tools, this PR removes them.

Test Plan:
regression tests
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22856816](https://our.internmc.facebook.com/intern/diff/D22856816)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fix] Remove activation_post_process in qat modules"


Summary:
Currently activation_post_process are inserted by default in qat modules, which is not
friendly to automatic quantization tools, this PR removes them.

Test Plan:
regression tests
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22856816](https://our.internmc.facebook.com/intern/diff/D22856816)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fix] Remove activation_post_process in qat modules"


Summary:
Currently activation_post_process are inserted by default in qat modules, which is not
friendly to automatic quantization tools, this PR removes them.

Test Plan:
regression tests
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22856816](https://our.internmc.facebook.com/intern/diff/D22856816)

[ghstack-poisoned]

zou3519 commented:
Reverting this PR because it appears to have broken some of the CUDA builds: https://circleci.com/gh/pytorch/pytorch/6684860?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link (on master)

The same test failure was present on this PR (https://circleci.com/gh/pytorch/pytorch/6665453?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) so it is likely this PR is the culprit.
